code/game/objects/items/knives.dm

```diff
@@ -124,18 +124,41 @@
 	name = "combat knife"
 	icon = 'icons/obj/weapons/stabby.dmi'
 	icon_state = "buckknife"
+	worn_icon_state = "buckknife"
 	desc = "A military combat utility survival knife."
 	embedding = list("pain_mult" = 4, "embed_chance" = 65, "fall_chance" = 10, "ignore_throwspeed_threshold" = TRUE)
 	force = 20
 	throwforce = 20
 	attack_verb_continuous = list("slashes", "stabs", "slices", "tears", "lacerates", "rips", "cuts")
 	attack_verb_simple = list("slash", "stab", "slice", "tear", "lacerate", "rip", "cut")
 	bayonet = TRUE
+	slot_flags = ITEM_SLOT_MASK
+
+/obj/item/knife/combat/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/knockoff, 90, list(BODY_ZONE_PRECISE_MOUTH), slot_flags) //90% to knock off when wearing a mask
+
+/obj/item/knife/combat/dropped(mob/living/user, slot)
+	. = ..()
+	if(user.get_item_by_slot(ITEM_SLOT_MASK) == src && prob(20))
```

Comment:
```suggestion
	if(user.get_item_by_slot(ITEM_SLOT_MASK) == src && !user.has_status_effect(/datum/status_effect/choke) && prob(20))
```
Another nit, since swallowing the knife also **drops** it inside the mob.